integration/update_test.go

```diff
@@ -466,6 +466,58 @@ func TestUpdateSet(t *testing.T) {
 	}
 }
 
+func TestUpdateTimestamp(t *testing.T) {
+	t.Parallel()
+
+	// store the current timestamp with $currentDate operator;
+	t.Run("currentDateReadBack", func(t *testing.T) {
+		maxDifference := time.Duration(2 * time.Second)
+		now := time.Now()
+		nowTimestamp := primitive.Timestamp{T: uint32(now.Unix()), I: uint32(0)}
+		id := "string-empty"
+
+		update := bson.D{{"$currentDate", bson.D{{"value", bson.D{{"$type", "timestamp"}}}}}}
+		stat := &mongo.UpdateResult{
+			MatchedCount:  1,
+			ModifiedCount: 1,
+			UpsertedCount: 0,
+		}
+		path := types.NewPathFromString("value")
+		result := bson.D{{"_id", id}, {"value", nowTimestamp}}
+
+		ctx, collection := setup(t, shareddata.Scalars, shareddata.Composites)
```

Comment:
I think it's better to move this line to the top of the test as we do with all other tests.